python/cudf/cudf/core/dataframe.py

```diff
@@ -357,6 +357,11 @@ def _getitem_tuple_arg(self, arg):
                     # as join is not assigning any names to index,
                     # update it over here
                     df.index.name = columns_df.index.name
+                    if not columns_df._data.multiindex and is_numeric_dtype(
```

Comment:
Curious why it matters checking `not columns_df._data.multiindex` since this is just `astype`ing the `index` not the `columns`?

Reply:
Apologies, it was to guard against `df.multiindex`